src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -359,6 +361,18 @@ export const extractMyInfoCookie = (
   return err(new MyInfoMissingAccessTokenError())
 }
 
+/**
+ * Checks if myInfoCookie is successful and returns the result.
+ * This function acts as a discriminator so that the type of the cookie is encoded in its type
+ * @param cookie the cookie to
+ * @returns ok(cookie) the successful myInfoCookie
+ * @returns err(cookie) the errored cookie
+ */
+export const extractSuccessfulCookie = (
+  cookie: MyInfoCookiePayload,
+): Result<MyInfoSuccessfulCookiePayload, MyInfoErroredCookiePayload> =>
```

Comment:
Should we change to returning an error? Feels unnatural to return the same cookie, but as an error type.